tests/PHPStanTests/nsrt/preg-match.php

```diff
@@ -15,25 +15,33 @@ function doMatch(string $s): void
     assertType('array{}|array{string}', $matches);
 
     if (Preg::match('/Price: (£|€)\d+/', $s, $matches)) {
-        assertType('array{string, string|null}', $matches);
+        assertType('array{string, string}', $matches);
     } else {
         assertType('array{}', $matches);
     }
-    assertType('array{}|array{string, string|null}', $matches);
+    assertType('array{}|array{string, string}', $matches);
 
     if (Preg::match('/Price: (£|€)?\d+/', $s, $matches)) {
-        assertType('array{0: string, 1?: string|null}', $matches);
+        assertType('array{0: string, 1: string|null}', $matches);
     } else {
         assertType('array{}', $matches);
     }
-    assertType('array{}|array{0: string, 1?: string|null}', $matches);
+    assertType('array{}|array{0: string, 1: string|null}', $matches);
+
+    // passing the PREG_UNMATCHED_AS_NULL should change nothing compared to above as it is always set
+    if (Preg::match('/Price: (£|€)?\d+/', $s, $matches, PREG_UNMATCHED_AS_NULL)) {
+        assertType('array{0: string, 1: string|null}', $matches);
+    } else {
+        assertType('array{}', $matches);
+    }
+    assertType('array{}|array{0: string, 1: string|null}', $matches);
 
     if (Preg::isMatch('/Price: (?<currency>£|€)\d+/', $s, $matches)) {
-        assertType('array{0: string, currency: string|null, 1: string|null}', $matches);
```

Comment:
since `PREG_UNMATCHED_AS_NULL` is implied, I think the expectations were correct before this PR

Reply:
Nope as the group is not optional it cannot be null here, this regex is missing the `?` from above, but this one does not error I think so all good.